### PR TITLE
Use box

### PR DESCRIPTION
**Pull request by odedre**
## Proposed Changes

- added use box and all its internal components
- using use box in component page top bar

### CODE REVIEW

File: scopes/ui-foundation/use-box/menu/menu.module.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  padding: 0 24px;
+  line-height: 14px;
+  border-bottom: 1px solid var(--border-color, #ededed);
+  .yarnTab {
+    color: #2c8ebb;
+    font-size: 16px;
+    padding-bottom: 6px;
+  }
+  .npmTab {
+    color: #c7363a;
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.module.scss:46]: npm brand color? extract to a scss var in the same file

File: scopes/ui-foundation/use-box/menu/menu.module.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  }
+}
+
+.tabs {
+  display: flex;
+  padding: 0 24px;
+  line-height: 14px;
+  border-bottom: 1px solid var(--border-color, #ededed);
+  .yarnTab {
+    color: #2c8ebb;
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.module.scss:41]: yarn brand color? extract to a scss var in the same file

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  /**
+   * component name to be presented
+   */
+  componentName: string;
+} & React.HTMLAttributes<HTMLDivElement>;
+
+export function Menu({ packageLink, bitLink, registryName, componentName }: MenuProps) {
+  const [activeTab, setActiveTab] = useState('bit');
+  const [activeRegistry, setActiveRegistry] = useState<string | undefined>(undefined);
+  if (activeRegistry === 'import') {
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.tsx:34]: space out

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      <div className={styles.top}>
+        <div className={styles.title}>
+          <Icon of="terminal" />
+          <span>{`Use ${componentName}`}</span>
+        </div>
+      </div>
+      <Tabs activeTab={activeTab} onClick={setActiveTab} />
+      {(activeTab === 'bit' || !activeTab) && (
+        <Import
+          // componentName={componentName}
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.tsx:58]: ?

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { Icon } from '@teambit/evangelist.elements.icon';
+import { BitInfo } from '@teambit/ui-foundation.ui.use-box.bit-info';
+import { Registry } from './registry';
+import { Install } from './install';
+import { Import } from './import';
+import styles from './menu.module.scss';
+import { Tabs } from './tabs';
+
+// export type TabOptions = "registry" | "import" | "install";
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.tsx:10]: cleanup

File: scopes/ui-foundation/use-box/menu/tabs.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Tab } from '@teambit/ui-foundation.ui.use-box.tab';
+import styles from './menu.module.scss';
+
+export function Tabs({ onClick, activeTab }: { activeTab: string; onClick: (active: string) => void }) {
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/tabs.tsx:5]: can set the default value to be `activeTab = 'bit'`

File: scopes/ui-foundation/use-box/scope-menu/scope-menu.module.scss [hunk 7]
```diff
@@ -0,0 +1,27 @@
+.top {
+  padding: 15px 24px;
+  color: var(--text-color);
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/scope-menu/scope-menu.module.scss:3]:
```suggestion
  color: var(--bit-text-color, #414141);
```

File: scopes/ui-foundation/use-box/scope-menu/scope-menu.module.scss [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+.title {
+  display: flex;
+  align-items: center;
+  font-size: var(--p-xs);
+  font-weight: bold;
+  > span {
+    &:first-child {
+      color: var(--text-highlight);
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/scope-menu/scope-menu.module.scss:13]:
```suggestion
      color: var(--bit-text-color-heavy, #2b2b2b);
```

File: scopes/ui-foundation/use-box/scope-menu/scope-menu.module.scss [hunk 9]
```diff
@@ -0,0 +1,27 @@
+.top {
+  padding: 15px 24px;
+  color: var(--text-color);
+}
+
+.title {
+  display: flex;
+  align-items: center;
+  font-size: var(--p-xs);
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/scope-menu/scope-menu.module.scss:9]:
```suggestion
  font-size: var(--bit-p-xs, 14px);
```

File: scopes/ui-foundation/use-box/scope-menu/scope-menu.module.scss [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  font-weight: bold;
+  > span {
+    &:first-child {
+      color: var(--text-highlight);
+      font-weight: bold;
+      margin-right: 7px;
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/scope-menu/scope-menu.module.scss:15]:
```suggestion
      margin-right: 8px;
```

File: scopes/ui-foundation/use-box/tab-content/tab-content.module.scss [hunk 11]
```diff
@@ -0,0 +1,12 @@
+.middle {
+  // margin-bottom: 15px;
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/tab-content/tab-content.module.scss:2]: cleanup

File: scopes/ui-foundation/use-box/tab/tab.module.scss [hunk 12]
```diff
@@ -0,0 +1,13 @@
+.tab {
+  margin-right: 8px;
+  padding: 0 16px 8px 16px;
+  font-size: 0.85em;
+  font-weight: bold;
+  text-transform: uppercase;
+  user-select: none;
+  cursor: pointer;
+  &.active {
+    border-bottom: 3px solid var(--base-color);
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/tab/tab.module.scss:10]:
```suggestion
    border-bottom: 3px solid var(--bit-accent-color, #6c5ce7);
```

File: scopes/ui-foundation/use-box/tab/tab.module.scss [hunk 13]
```diff
@@ -0,0 +1,13 @@
+.tab {
+  margin-right: 8px;
+  padding: 0 16px 8px 16px;
+  font-size: 0.85em;
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/tab/tab.module.scss:4]: <3

**Review by KutnerUri — APPROVED** [review]

File: scopes/ui-foundation/use-box/back-button/back.tsx [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Icon } from '@teambit/evangelist.elements.icon';
+import styles from './back.module.scss';
+
+export type BackProps = {
+  setActive: (active: string) => void;
+  prevTab: string;
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/back-button/back.tsx:8]: no need for this. just passed `onClick`

File: scopes/ui-foundation/use-box/back-button/back.module.scss [hunk 15]
```diff
@@ -0,0 +1,13 @@
+.back {
+  border-bottom: 1px solid var(--border-color);
+  padding: 15px 24px;
+  > div {
+    display: flex;
+    align-items: center;
+    cursor: pointer;
+    width: fit-content;
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/back-button/back.module.scss:8]: its so the click on 'back' wont span on the entire line and will only work if actually clicking the text and arrow

File: scopes/ui-foundation/use-box/back-button/back.module.scss [hunk 16]
```diff
@@ -0,0 +1,13 @@
+.back {
+  border-bottom: 1px solid var(--border-color);
+  padding: 15px 24px;
+  > div {
+    display: flex;
+    align-items: center;
+    cursor: pointer;
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/back-button/back.module.scss:7]: no. I dont want to have to override the button default css

File: scopes/ui-foundation/use-box/dropdown/dropdown.composition.tsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+  width: '400px',
+};
+
+export const UseBoxExample = () => {
+  return (
+    <ThemeCompositions>
+      <div style={styles}>
+        <UseBoxDropdown
+          defaultActiveOption="bit"
+          Menu={() => (
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/dropdown/dropdown.composition.tsx:17]: I get a ts error
```
JSX element type 'Menu' does not have any construct or call signatures.ts(2604)

```

File: scopes/ui-foundation/use-box/dropdown/dropdown.module.scss [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+.menu {
+  padding: 0;
+  right: 0;
+  font-size: var(--p-xs);
+  border-radius: 6px;
+  width: 400px;
+}
+
+.placeholder {
+  color: #37b26c;
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/dropdown/dropdown.module.scss:18]: ?

File: scopes/ui-foundation/use-box/dropdown/dropdown.module.scss [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+  border-radius: 6px;
+  user-select: none;
+  transition: background-color 300ms, color 300ms ease-in-out;
+  &:hover {
+    background-color: var(--bg-dent);
+    color: white;
+    background-color: #2e945a;
+    border-color: #2e945a;
+  }
+  @media screen and (max-width: 768px) {
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/dropdown/dropdown.module.scss:36]: cant use it. doesnt have any index file

File: scopes/ui-foundation/use-box/dropdown/dropdown.tsx [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+
+export type UseBoxDropdownProps = {
+  Menu?: ComponentType;
+  containerClass?: string;
+  defaultActiveOption?: string;
+} & React.HTMLAttributes<HTMLDivElement>;
+
+export function UseBoxDropdown({ className, Menu, ...rest }: UseBoxDropdownProps) {
+  const [activeTab, setActiveTab] = useState(0);
+  if (!Menu) return null;
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/dropdown/dropdown.tsx:14]: dont know if I want to show the placeholder without the menu

[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/dropdown/dropdown.tsx:14]: I just made it required so now we dont need all this

File: scopes/ui-foundation/use-box/dropdown/dropdown.tsx [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <div className={classNames(styles.dropdown, className)} {...rest}>
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/dropdown/dropdown.tsx:16]: I need to figure out how to pass the margin between the placeholder and the menu. very annoying.
lets talk about it

File: scopes/ui-foundation/use-box/menu/import.tsx [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+        <div className={linkStyles} onClick={back}>
+          <Icon of="download" />
+          <span>Install Bit on your computer</span>
+        </div>
+      }
+    >
+      <div className={styles.importContent}>
+        <div>Add heading as a dependency</div>
+        <CopyBox>{`bit install ${packageName}`}</CopyBox>
+        <div>Import heading to your workspace</div>
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/menu/import.tsx:37]: 10x

File: scopes/ui-foundation/use-box/menu/import.tsx [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+      {...rest}
+      bottom={
+        <div className={linkStyles} onClick={back}>
+          <Icon of="download" />
+          <span>Install Bit on your computer</span>
+        </div>
+      }
+    >
+      <div className={styles.importContent}>
+        <div>Add heading as a dependency</div>
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/menu/import.tsx:35]: 10x

File: scopes/ui-foundation/use-box/menu/install.tsx [hunk 24]
```diff
@@ -0,0 +1,38 @@
...
+            <Icon of="settings" />
+            <span>
+              Configure <HighlightedText size="xxs">{registryName}</HighlightedText> as a Scoped Registry
+            </span>
+          </div>
+          <Icon of="arrow_right" />
+        </div>
+      }
+    >
+      <div>{`Install ${componentName} with a ${packageManager}`}</div>
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/ui-foundation/use-box/menu/install.tsx:34]: 10x

**Pull request merged**